File: C2L/LuaCallStatic.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

namespace JP
{
    public static class TestStatic
    {
        public static int ID = 99;
        public static string Name
        {
            get;
            set;
        }
        public static string OutPut()
        {
            return "static";
        }
        public static void  Default(string str="abc")
        {
            Debug.Log(str);
        }
    }
}
public class LuaCallStatic : MonoBehaviour
{
    
    void Start()
    {
        xLuaEnv.Instance.Dostring("require('C2L/LuaCallStatic')");
    }
    private void OnDestroy()
    {
        xLuaEnv.Instance.Free();
    }


}

File: DataPath/Lua/C2L/LuaCallStatic.lua
--print('LuaCallStatic')
-- Lua调用静态类
--规则“CS.命名空间.类名.成员名”
print(CS.JP.TestStatic.ID)

--给静态属性赋值
CS.JP.TestStatic.Name="admin"
print(CS.JP.TestStatic.Name)
print(CS.JP.TestStatic.ID)

--静态成员方法调用
print(CS.JP.TestStatic.OutPut())
--调用Default使用默认值
CS.JP.TestStatic.Default()
--使用lua的值
CS.JP.TestStatic.Default("def")